### PR TITLE
WP-561-menu-items-translation

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

**Review by dimitrystd — CHANGES_REQUESTED** [review]
Let's invert logic. Instead of pass "flag" through pipeline we should change all callers to call pipeline (get related items) only if it's required.

File: js/smartling-connector-admin.js [hunk 1]
```diff
@@ -284,14 +284,22 @@ function ajaxDownload() {
                         }
                         break;
                 }
-                if (wp && wp.data && wp.data.dispatch && wp.data.dispatch('core/notices').createSuccessNotice && wp.data.dispatch('core/notices').createErrorNotice) {
```
Comment (PavelLoparev) [inline js/smartling-connector-admin.js:287]: I remember it was added in previous PR. Why delete this now?

**Review by PavelLoparev — COMMENTED** [review]
In general, seems like  you just added one condition into each content entity type upload flow.

Probably the question are all the existing tests passed? And do we have test for menu items specifically?

**Comment by vsolovei-smartling** [comment]
> In general, seems like you just added one condition into each content entity type upload flow.
> 
> Probably the question are all the existing tests passed? And do we have test for menu items specifically?

Except the one uncoditional call, yes.
Test pass, I'm currently trying to implement a menu items test

File: js/smartling-connector-admin.js (hunk 1 above)
Comment (vsolovei-smartling) [reply js/smartling-connector-admin.js:287]: The client in question reported it only makes matters worse for them

**Review by dimitrystd — APPROVED** [review]

**Review by dimitrystd — APPROVED** [review]

**Pull request merged**